evm/src/executor/inspector/cheatcodes/env.rs

```diff
@@ -156,6 +156,25 @@ pub fn apply<DB: Database>(
             Ok(Bytes::new())
         }
         HEVMCalls::Accesses(inner) => Ok(accesses(state, inner.0)),
+        HEVMCalls::SetNonce(inner) => {
+            // TODO:  this is probably not a good long-term solution since it might mess up the gas
+            // calculations
+            data.subroutine.load_account(inner.0, data.db);
+
+            // we can safely unwrap because `load_account` insert inner.0 to DB.
+            let account = data.subroutine.state().get_mut(&inner.0).unwrap();
+            // nonce must increment only
+            if account.info.nonce < inner.1 {
+                account.info.nonce = inner.1;
+                Ok(Bytes::new())
+            } else {
+                Err("Nonce too low.".to_string().encode().into())
+            }
+        }
+        HEVMCalls::GetNonce(inner) => match data.subroutine.state().get(&inner.0) {
+            Some(account) => Ok(abi::encode(&[Token::Uint(account.info.nonce.into())]).into()),
+            None => Err("Account not in state DB.".to_string().encode().into()),
```

Comment:
Here we should use whatever the nonce is after `load_account` for the same reasons as above